robolectric-resources/src/main/java/org/robolectric/manifest/AndroidManifest.java

```diff
@@ -152,7 +152,14 @@ void parseAndroidManifest() {
         }
 
         minSdkVersion = getTagAttributeIntValue(manifestDocument, "uses-sdk", "android:minSdkVersion");
-        targetSdkVersion = getTagAttributeIntValue(manifestDocument, "uses-sdk", "android:targetSdkVersion");
+
+        String targetSdkText = getTagAttributeText(manifestDocument, "uses-sdk",
+            "android:targetSdkVersion");
+        if (targetSdkText != null) {
+          // Support Android O Preview. This can be removed once Android O is officially launched.
+          targetSdkVersion = targetSdkText.equals("O") ? 26 : Integer.parseInt(targetSdkText);
```

Comment:
I think we'll want to change this to make it future proof. All preview versions are going to be code words, so lets just return the string value and process it outside of this class, if it fails to parse then we choose the previous highest SDK value. The version number returned here should be 25 actually.